### PR TITLE
More strict typing [#180206625]

### PR DESCRIPTION
**Pull request by olzraiti**
https://www.pivotaltracker.com/story/show/180206625
https://211119.dev.laji.fi

### CODE REVIEW

File: projects/laji/src/app/shared/service/id.service.ts [hunk 1]
```diff
@@ -31,14 +31,16 @@ export class IdService {
     return value;
   }
 
-  static getUri(value) {
+  static getUri(value: any) {
```
Comment (Blodir) [inline projects/laji/src/app/shared/service/id.service.ts:34]: same as above

File: projects/laji/src/app/shared/service/laji-api.interceptor.ts [hunk 2]
```diff
@@ -87,14 +86,14 @@ export class LajiApiInterceptor implements HttpInterceptor {
 
     setTimeout(() => {
       if (request.body instanceof FormData) {
-        const data = [];
+        const data: any[] = [];
```
Comment (Blodir) [inline projects/laji/src/app/shared/service/laji-api.interceptor.ts:89]: `const data: Observable<IFileLoad>[] = [];`
(`IFileLoad` also needs to be exported)

File: projects/laji/src/app/shared/service/laji-api.service.ts [hunk 3]
```diff
@@ -278,15 +278,15 @@ export class LajiApiService {
     );
   }
 
-  private removeUndefinedFromObject(obj: object) {
+  private removeUndefinedFromObject<T extends object>(obj: T): T {
```
Comment (Blodir) [inline projects/laji/src/app/shared/service/laji-api.service.ts:281]: Doesn't this line contradict the next one? If T extends object, then there should be no need to check whether obj is an object or not?

**Review by Blodir — CHANGES_REQUESTED** [review]

File: projects/laji/src/app/shared/service/id.service.ts [hunk 4]
```diff
@@ -10,15 +10,15 @@ export const DEFAULT_DOMAIN = 'http://tun.fi/';
...
+  static readonly domainMap: Record<string, string> = {
     'luomus:': 'http://id.luomus.fi/',
     'zmuo:': 'http://id.zmuo.oulu.fi/',
     'herbo:': 'http://id.herb.oulu.fi/',
     'utu:': 'http://mus.utu.fi/',
     'gbif-dataset:': 'https://www.gbif.org/dataset/',
   };
 
-  static getId(value) {
+  static getId(value: any) {
```
[the remark this answers is not in the document]
Comment (olzraiti) [reply projects/laji/src/app/shared/service/id.service.ts:21]: Yeah I tried to refactor this first, but it wasn't very easy and I'd rather not break stuff. The generic is a good idea, I'll apply that

**Comment by olzraiti** [comment]
Thanks for the review! I addressed the issues.

**Review by Blodir — APPROVED** [review]

**Pull request merged**